daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/DelimiterUnparsers.scala

```diff
@@ -54,6 +55,15 @@ class DelimiterTextUnparser(
       s"Unparsing starting at bit position: ${state.dataOutputStream.maybeAbsBitPos0b}",
     )
 
+    // Do not emit the delimiter if the value is empty and the emptyValueDelimiterPolicy
+    // indicates it should be suppressed
+    if (state.currentInfosetNode.isSimple) {
+      val node = state.currentInfosetNode.asSimple
+      if (node.hasValue && node.isEmpty && !requiredOnEmptyValue) {
+        return
```

Comment:
We usually try to avoid return. I think the only way around it is to wrap everyhing in an if-block, which could be messy. So maybe it's fine in this case? Thoughts from others?

Reply:
Good point.  The whole function is about 60 lines long and the return occurs 10 lines into the function's body.  The rest of the function would have to be wrapped in an if-block, or the function refactored to call sub-functions which do the rest of the work.  I'm fine with the return if others agree.

Reply:
Return is "maintenance unstable" in scala.  You modify the code at some point and an exception gets involved and then the return ends up causing real problems. 

Best to remove them. They don't compose well with exception throw/catch/try or Try objects.